### ISSUE TITLE
install_generator.rb ngrok regex is not matching all possible subdomains

### ISSUE THREAD

**Issue by lonroth** [Shopify/shopify_app#1182]
The current `install_generator.rb` (https://github.com/Shopify/shopify_app/blob/f74f00c2ce053ea3160571a705fdd1b90969f119/lib/generators/shopify_app/install/install_generator.rb#L68) ngrok code is *not matching* all possible subdomains:
```
config.hosts = (config.hosts rescue []) << /\w+\.ngrok\.io/
```
As soon as you have a ngrok subdomain that includes other characters for eg `-` (which is a correct subdomain) it doesn't catch these ngrok domains with the current regex. The regex should be updated so it matches all valid ngrok subdomains.

**Comment by resistorsoftware** [comment]
Note too that while discouraged, subdomains with an underscore are also legal but not with Shopify!

**Comment by lonroth** [comment]
@resistorsoftware Good detail that the underscore is not allowed on Shopify and then it make sense to omit this version of a subdomain. However I still believe it would be good to support other versions out of the box such as `-` and not limit it to solely `\w` ? I don't use randomly assigned ngrok names but rather one dedicated per Shopify App (however I use `-` in the names of those apps). Of course I just overwrite this setting when I start a new Rails app with the GEM so it's a quick fix but still would be nice if it wasn't needed for people installing the GEM for the first time and generating the files. I can probably do a quick PR for this unless we decide it's not valuable enough and if so we can close this issue.

**Comment by resistorsoftware** [comment]
Exactly. In my initial steps, the drudgery always involves first making an entry in .puma-dev directory for an App with a name like my-app or hooligan-scotch with a port number. That way I can rely on https://my-app.test or https://hooligan-scotch.test being available with a port for Puma. Then I teach my puma config.rb that an ngrok tunnel will be needed. The final step is to edit .env with the Shopify API credentials, and Ngrok settings for https://my-app.test and any webhooks that would subsequently be directed to https://my-app.ngrok.io or https://hooligan-scotch.ngrok.io. 

So when I fire up that Procfile and Rails begins serving up an App, if I did anything wrong, it's obvious. Then the chore is to attack the environment, specifically as you say, the https hosts so that the ngrok requests pass in AND the my-app.test ones too. I just <<  my-app.test as well, so things play nice. 

So many ways Ngrok, puma-ngrok-tunnel and Puma dev make Shopify App work super pleasant. I could go on and on with hot tips and tricks but will refrain for now.
